py-polars/Makefile

```diff
@@ -91,53 +93,55 @@ clippy:  ## Run clippy
+test: $(VENV)  ## Run fast unittests
+	$(VENV_ACT_BIN)pytest -v tests/unit/
 
 .PHONY: doctest
-doctest: $(VENV) develop  ## Run doctests
+doctest: $(VENV)  ## Run doctests
 	$(VENV_ACT_BIN)python tests/docs/run_doc_examples.py
 
 .PHONY: test-all
-test-all: # develop  ## Run all tests
-	$(VENV_ACT_BIN)pytest
-	$(VENV_ACT_BIN)python tests/docs/run_doc_examples.py
+test-all: $(VENV)  ## Run all tests
+	-$(VENV_ACT_BIN)pytest -v
+	-$(VENV_ACT_BIN)python tests/docs/run_doc_examples.py
 
 .PHONY: coverage
-coverage:  ## Run tests and report coverage
-	$(VENV_ACT_BIN)pytest --cov
+coverage: $(VENV)  ## Run tests and report coverage
+	$(VENV_ACT_BIN)pytest -v --cov
 
 build-debug: $(VENV)  ## Build debug wheel
 	$(VENV_ACT_BIN)maturin build -o $(WHEELS_DEBUG)
 
 build: $(VENV)  ## Build release wheel
-	$(VENV_ACT_BIN)maturin build -o $(WHEELS_RELEASE) --release -- -C codegen-units=16 -C lto=thin -C target-cpu=native
+	$(VENV_ACT_BIN)maturin build -o $(WHEELS_RELEASE) $(RELEASE_FLAGS)
 
 install-debug: $(VENV)  ## pip install debug wheel
 	$(VENV_ACT_BIN)pip install --no-deps --force-reinstall -U $(WHEELS_DEBUG)/polars-*.whl
 
 install: $(VENV)  ## pip install release wheel
 	$(VENV_ACT_BIN)pip install --no-deps --force-reinstall -U $(WHEELS_RELEASE)/polars-*.whl
 
-# Shortcut targets
-build-install-debug: build-debug install-debug  ## Build and install debug wheel
-build-install: build install  ## Build and install release wheel
-
 .PHONY: help
 help:  ## Display this help screen
 	@echo "Usage:"
-	@echo "* To run a make <comand> with venv, issue"
-	@echo "  > make <command>"
-	@echo "* To run a make <command> with alternative environement <name>, issue"
-	@echo "  > make VENV=<name> <command>"
-	@echo "* To run a make <command> without virtual environment (ex: conda), issue"
-	@echo "  > make VENV='' <command>"
+	@echo "  * To run a make <comand> with venv, issue"
+	@echo "    $$ make <command>"
+	@echo "  * To run a make <command> with alternative environement <name>, issue"
+	@echo "    $$ make VENV=<name> <command>"
+	@echo "  * To run a make <command> without virtual environment (ex: conda), issue"
+	@echo "    $$ make VENV='' <command>"
 	@echo
-	@echo "Currently:"
-	@echo "  VENV = '$(VENV)'"
-	@echo
-
 	@echo -e '\033[1mAvailable commands:\033[0m'
 	@grep -E '^[a-z.A-Z_-]+:.*?## .*$$' $(MAKEFILE_LIST) | awk 'BEGIN {FS = ":.*?## "}; {printf "  \033[36m%-18s\033[0m %s\n", $$1, $$2}' | sort
+	@echo
+	@echo "Examples:"
+	@echo "  * Compile polars (debug) in venv and run test-all in one go"
+	@echo "    $$ make develop test-all"
+	@echo "  * Build polars release wheel, install it in venv and run test-all in one go"
+	@echo "    $$ make build install test-all"
+	@echo "  * Build release wheel, then install into venv and outside of venv"
+	@echo "    $$ make build   # build release wheel"
+	@echo "    $$ make install # install release wheel into venv"
+	@echo "    $$ make install VENV='' # install release wheel outside of venv (e.g. in conda env)"
```

Comment:
I don't think we should include these examples. In general it feels like you're doing a lot with building wheels, but is that really what Polars developers are doing a lot?

This Makefile should help the 'general public' contribute effectively to Polars. Any highly specialized stuff should not be in here at all; you can just make your own console aliases for that kind of stuff.

The way I set up the Makefile was to include a few useful commands that anyone working with this code will want to use. Anything else should not be included because it increases the threshold for contributing ("What is all this stuff...? Nevermind.").

So please help me understand why your workflow includes these commands and why you think this is important for other contributors as well.

Reply:
> I don't think we should include these examples. In general it feels like you're doing a lot with building wheels, but is that really what Polars developers are doing a lot?

I am oblivious to these examples and can remove them. I only added them because you said earlier that the new Makefile is hard to read 😄    
Building wheels are essential for my workflow. We build, test in release mode, and deploy ourselves from those wheels.

> This Makefile should help the 'general public' contribute effectively to Polars. 

Totally agree! But providing a crippled Makefile does not serve the stated purpose. It should cover relevant build/test/install workflows of which there are several. If there are some pain-points, documentation (`make help`) can help to ease the burden. Please, suggest what else should be documented.

> Anything else should not be included because it increases the threshold for contributing ("What is all this stuff...? Nevermind.").

On the contrary, it is exactly the opposite. A framework designed for big data processing does not allow for release build testing or cannot make wheels for deployment -- "Not ready for production, nevermind, move on ..." 😄